src/commands/branch.rs

```diff
@@ -338,6 +345,45 @@ fn cmd_branch_list(
         }
     }
 
+    if !args.revisions.is_empty() {
+        // Match against local targets only, which is consistent with "jj git push".
+        fn local_targets(branch_target: &BranchTarget) -> &[CommitId] {
+            if let Some(target) = branch_target.local_target.as_ref() {
+                target.adds()
+            } else {
+                &[]
+            }
+        }
+
+        // Intersects with the set of all branch targets to minimize the lookup space.
```

Comment:
I would move this comment to right before `let all_targets=all_branches...` (line 365)

Reply:
Done